examples/voice-pipeline-agent/function_calling_weather.py

```diff
@@ -39,11 +40,25 @@ async def get_weather(
         # while awaiting the completion of the function call. To create a more dynamic and engaging
         # interaction, consider varying the responses based on context or user input.
         call_ctx = AgentCallContext.get_current()
-        message = f"Let me check the weather in {location} for you."
+        # message = f"Let me check the weather in {location} for you."
+        message = f"Here is the weather in {location}: "
+        filler_messages = [
+            "Let me check the weather in {location} for you.",
+            "Let me see what the weather is like in {location} right now.",
+            # LLM will complete this sentence if it is added to the end of the chat context
+            "The current weather in {location} is ",
+        ]
+        message = random.choice(filler_messages).format(location=location)
         speech_handle = await call_ctx.agent.say(message)  # noqa: F841
```

Comment:
Ideally, the say method should automatically do it, so the users don't have to do anything

Reply:
There might be other sources of `say` like from a callback of an event, we can't know if it's from a function call even there is a function call in progress right?

Reply:
AgentCallContext is using a ContextVar that is only available inside function calls

Reply:
I am thinking to get the call_ctx using `AgentCallContext.get_current` inside `say` method, but wondering is there a risk that there is a function call running while the user call `say` from other places like a callback of a event. In that case the `AgentCallContext.get_current` still returns a context?

Reply:
If that's possible, we can add an explicit arg like `add_to_call_ctx` (default False) for `say` so at least users can do that easier. What do you think?

Reply:
I don't think it is possible (at least very unlikely)

Reply:
Tho we can still add this new parameter

Reply:
@theomonnom Added a arg `add_to_fnc_call_ctx` for say, please take a look.

Reply:
I don't think so. `add_to_ctx` can be True for most of cases no matter it's inside or outside a function call.

Reply:
I think it's fine, you almost always want your new speech to be inside the context. And making it False will do what users expect to do

Reply:
Ok, I'll update to share the `add_to_ctx`

Reply:
Updated.